File: app/src/test/java/pl/laskowski/marcin/model/FieldTest.java
package pl.laskowski.marcin.model;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by Marcin Laskowski.
 */

public class FieldTest {

    @Test
    public void createEmptyField() {
        Field field = new Field(0, 0);

        assertTrue(field.isEmpty());
        assertEquals(0, field.value());
    }

    @Test
    public void setNewValueToField() {
        Field field = new Field(0, 0);
        assertTrue(field.isEmpty());
        field.set(7);
        assertEquals(7, field.value());
    }

    @Test
    public void createFieldWithCoords() {
        int x = 3;
        int y = 7;
        Field field = new Field(x, y);

        assertEquals(x, field.getX());
        assertEquals(y, field.getY());
    }

    @Test
    public void shouldBeEqualsIfCoordsAreEquals() {
        Field fieldBeingTested = new Field(1, 2);
        Field sameField = new Field(1, 2);
        Field differentField = new Field(3, 5);

        assertEquals(sameField, fieldBeingTested);
        assertNotEquals(differentField, fieldBeingTested);
    }

}

File: app/src/test/kotlin/com/github/ilikeyourhat/kudoku/model/FieldTest.kt
package com.github.ilikeyourhat.kudoku.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class FieldTest {

    @Test
    fun createEmptyField() {
        val field = Field(0, 0)
        assertTrue(field.isEmpty)
        assertEquals(0, field.value)
    }

    @Test
    fun setNewValueToField() {
        val field = Field(0, 0)
        assertTrue(field.isEmpty)

        field.set(7)
        assertEquals(7, field.value)
    }

    @Test
    fun createFieldWithCoords() {
        val x = 3
        val y = 7
        val field = Field(x, y)
        assertEquals(x, field.x)
        assertEquals(y, field.y)
    }

    @Test
    fun shouldBeEqualsIfCoordsAreEquals() {
        val fieldBeingTested = Field(1, 2)
        val sameField = Field(1, 2)
        val differentField = Field(3, 5)

        assertEquals(sameField, fieldBeingTested)
        assertNotEquals(differentField, fieldBeingTested)
    }
}